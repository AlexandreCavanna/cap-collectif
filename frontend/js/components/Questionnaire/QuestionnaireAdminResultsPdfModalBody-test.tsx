import * as React from 'react'
import { shallow } from 'enzyme'
import QuestionnaireAdminResultsPdfModalBody from './QuestionnaireAdminResultsPdfModalBody'

describe('<QuestionnaireAdminResultsPdfModalBody />', () => {
  const props = {
    loading: false,
    error: false,
    retryCount: 0,
  }
  const loadingProps = { ...props, loading: true }
  const errorProps = { ...props, loading: false, error: true }
  const retryCountExceedLimitProps = { ...errorProps, retryCount: 1 }
  it('renders correctly', () => {
    const wrapper = shallow(<QuestionnaireAdminResultsPdfModalBody {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('renders correctly when loading', () => {
    const wrapper = shallow(<QuestionnaireAdminResultsPdfModalBody {...loadingProps} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('renders correctly when error', () => {
    const wrapper = shallow(<QuestionnaireAdminResultsPdfModalBody {...errorProps} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('renders correctly when retryCount exceed limit', () => {
    const wrapper = shallow(<QuestionnaireAdminResultsPdfModalBody {...retryCountExceedLimitProps} />)
    expect(wrapper).toMatchSnapshot()
  })
})
