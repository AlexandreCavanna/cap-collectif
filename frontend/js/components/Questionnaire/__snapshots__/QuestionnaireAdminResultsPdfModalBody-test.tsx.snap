// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionnaireAdminResultsPdfModalBody /> renders correctly 1`] = `
<FormattedMessage
  id="data.exported.to.pdf"
  values={Object {}}
/>
`;

exports[`<QuestionnaireAdminResultsPdfModalBody /> renders correctly when error 1`] = `
<FormattedMessage
  id="sorry.export.failed"
  values={Object {}}
/>
`;

exports[`<QuestionnaireAdminResultsPdfModalBody /> renders correctly when loading 1`] = `
<FormattedMessage
  id="this.may.take.a.few.moments"
  values={Object {}}
/>
`;

exports[`<QuestionnaireAdminResultsPdfModalBody /> renders correctly when retryCount exceed limit 1`] = `
<FormattedMessage
  id="sorry.export.failed.try.later"
  values={Object {}}
/>
`;
